### PR TITLE
[BE] Fix SIM118 `in-dict-keys`

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100336
* #100342
* #100341
* #100338
* __->__ #100339

Use {key} in {dict} instead of {key} in {dict}.keys()

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @anijain2305 @soumith @desertfire

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -52,7 +52,6 @@ ignore = [
     "SIM115",
     "SIM116", # Disable Use a dictionary instead of consecutive `if` statements
     "SIM117",
-    "SIM118",
```
Comment (Skylion007) [inline pyproject.toml:55]: Please edit the flake8 config as well.

File: torch/profiler/_utils.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def __init__(self, prof: profile):
         self.profile = prof
         self.metrics: Dict[EventKey, EventMetrics] = {}
         self.compute_self_time()
-        self.event_keys = sorted((e for e in self.metrics.keys()),
+        self.event_keys = sorted((e for e in self.metrics),
```
Comment (Skylion007) [inline torch/profiler/_utils.py:99]:
```suggestion
        self.event_keys = sorted(self.metrics,
```

File: pyproject.toml (hunk 1 above)
Comment (justinchuby) [reply pyproject.toml:55]: Done

File: torch/profiler/_utils.py (hunk 2 above)
Comment (justinchuby) [reply torch/profiler/_utils.py:99]: Done

Comment (Skylion007) [reply torch/profiler/_utils.py]: I just opened an issue to add a new flake8 comprehension rule to detect these. :)

Comment (justinchuby) [reply torch/profiler/_utils.py]: There probably is one already IIRC. I am not sure if it is enabled in our config

Comment (Skylion007) [reply torch/profiler/_utils.py]: @justinchuby Nah, I enabled almost every flake8-comprehension rule in our config, even the ones I added recently. :)

File: test/cpp/api/optim_baseline.py
Comment (janeyx99) [inline test/cpp/api/optim_baseline.py:1]: changes in this file LGTM

File: test/test_optim.py
Comment (janeyx99) [inline test/test_optim.py:1]: Also LGTM

File: torch/optim/swa_utils.py [hunk 3]
```diff
@@ -269,7 +269,7 @@ def update_bn(loader, model, device=None):
 
         model(input)
 
-    for bn_module in momenta.keys():
+    for bn_module in momenta:
```
Comment (janeyx99) [inline torch/optim/swa_utils.py:272]: Would it be better to use an items() call here since we access the key in the next line?

**Review by NivekT — COMMENTED** [review]
Changes under `torch/utils/data/datapipes/iter`LGTM!

**Comment by albanD** [comment]
Hey!
I think it would be better to split this kind of PRs into smaller chunks (since you're already using ghstack, that shouldn't be a problem).
That will make reviewing a lot easier and also allow you to pinpoint the actual breakage from CI more easily.

**Pull request closed without merging**